storage/decorator.go

```diff
@@ -0,0 +1,42 @@
+package storage
```

Comment:
@yurishkuro let me know if you'd like this decorator to live elsewhere

Reply:
storage/storagemetrics